### PR TITLE
Mgmt 5399 Add support for conditions Failed Stopped

### PR DESCRIPTION
**Pull request by rollandf**

### CODE REVIEW

**Comment by rollandf** [comment]
/hold
on top of #1614

File: docs/kube-api-conditions.md [hunk 1]
```diff
@@ -26,6 +26,14 @@ Assisted Installer handles additional types: `SpecSynced`, `ReadyForInstallation
...
 |Installed|False|InstallationNotStarted|The installation has not yet started|If the cluster is before installation ("insufficient"/"pending-for-input"/"ready")|
 |Installed|False|InstallationInProgress|The installation is in progress: "status_info"|If the cluster is installing ("preparing-for-installation", "installing", "finalizing", "installing-pending-user-action")|
+||||||
+|Failed|True|InstallationFailed|The installation failed: "status_info"|if the cluster status is "error"|
+|Failed|False|InstallationNotFailed|The installation has not failed|If the cluster status is not "error"|
+||||||
+|Stopped|True|InstallationFailed|The installation has stopped due to error|if the cluster status is "error"|
+|Stopped|True|InstallationCancelled|The installation has stopped because it was cancelled|if the cluster status is "cancelled"|
+|Stopped|True|InstallationCompleted|The installation has stopped because it completed successfully|if the cluster status is "installed"|
+|Stopped|False|InstallationNotStopped|The installation is in progress|If the cluster status is not "error", "cancelled" or "installed|
```
Comment (filanov) [inline docs/kube-api-conditions.md:36]: Regarding the message, do we separate between installation in progress (assisted service started the installation) and the discovery phase? or do we see all the process as installing? 
@avishayt @mhrivnak ?

**Comment by rollandf** [comment]
Conditions on ClusterDeployment:
```
status:
  cliImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:3574d6c1fcc46e1ebd41b7b887b92035ea18213133c06edad539cde25b767ff9
  conditions:
  - lastProbeTime: "2021-05-12T09:06:29Z"
    lastTransitionTime: "2021-05-12T09:06:29Z"
    message: Condition Initialized
    reason: Initialized
    status: Unknown
    type: Hibernating
  - lastProbeTime: "2021-05-12T09:06:29Z"
    lastTransitionTime: "2021-05-12T09:06:29Z"
    message: no ClusterRelocates match
    reason: NoMatchingRelocates
    status: "False"
    type: RelocationFailed
  - lastProbeTime: "2021-05-12T09:06:29Z"
    lastTransitionTime: "2021-05-12T09:06:29Z"
    message: Platform credentials passed authentication check
    reason: PlatformAuthSuccess
    status: "False"
    type: AuthenticationFailure
  - lastProbeTime: "2021-05-12T09:06:30Z"
    lastTransitionTime: "2021-05-12T09:06:30Z"
    message: ClusterImageSet test-infra-cluster-assisted-installer-image-set is available
    reason: ClusterImageSetFound
    status: "False"
    type: ClusterImageSetNotFound
  - lastProbeTime: "2021-05-12T09:06:43Z"
    lastTransitionTime: "2021-05-12T09:06:43Z"
    message: Images required for cluster deployment installations are resolved
    reason: ImagesResolved
    status: "False"
    type: InstallImagesNotResolved
  - lastProbeTime: "2021-05-12T09:06:43Z"
    lastTransitionTime: "2021-05-12T09:06:43Z"
    message: The installation has not failed
    reason: InstallationNotFailed
    status: "False"
    type: ClusterInstallFailed
  - lastProbeTime: "2021-05-12T09:20:09Z"
    lastTransitionTime: "2021-05-12T09:06:43Z"
    message: 'The installation is in progress: Finalizing cluster installation'
    reason: InstallationInProgress
    status: "False"
    type: ClusterInstallCompleted
  - lastProbeTime: "2021-05-12T09:06:43Z"
    lastTransitionTime: "2021-05-12T09:06:43Z"
    message: The installation is in progress
    reason: InstallationNotStopped
    status: "False"
    type: ClusterInstallStopped
  - lastProbeTime: "2021-05-12T09:07:39Z"
    lastTransitionTime: "2021-05-12T09:07:39Z"
    message: The cluster requirements are met
    reason: ClusterAlreadyInstalling
    status: "True"
    type: ClusterInstallRequirementsMet
  - lastProbeTime: "2021-05-12T09:06:43Z"
    lastTransitionTime: "2021-05-12T09:06:43Z"
    message: The installation has not failed
    reason: InstallationNotFailed
    status: "False"
    type: ProvisionFailed
  - lastProbeTime: "2021-05-12T09:06:43Z"
    lastTransitionTime: "2021-05-12T09:06:43Z"
    message: The installation is in progress
    reason: InstallationNotStopped
    status: "False"
    type: ProvisionStopped
  installStartedTimestamp: "2021-05-12T09:07:39Z"
  installVersion: 4.8.0-fc.3
  installerImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:e26b0d949145d33e53591f8a9f1fdf679f26d486ebac928f7f12655023eb1b7a

```

**Comment by rollandf** [comment]
Conditions on AgentClusterInstall:

```
Status:
  Conditions:
    Last Probe Time:             2021-05-12T09:06:30Z
    Last Transition Time:        2021-05-12T09:06:30Z
    Message:                     The Spec has been successfully applied
    Reason:                      SyncOK
    Status:                      True
    Type:                        SpecSynced
    Last Probe Time:             2021-05-12T09:07:39Z
    Last Transition Time:        2021-05-12T09:07:39Z
    Message:                     The cluster requirements are met
    Reason:                      ClusterAlreadyInstalling
    Status:                      True
    Type:                        RequirementsMet
    Last Probe Time:             2021-05-12T09:07:39Z
    Last Transition Time:        2021-05-12T09:07:39Z
    Message:                     The cluster's validations are passing
    Reason:                      ValidationsPassing
    Status:                      True
    Type:                        Validated
    Last Probe Time:             2021-05-12T09:20:09Z
    Last Transition Time:        2021-05-12T09:20:09Z
    Message:                     The installation is in progress: Finalizing cluster installation
    Reason:                      InstallationInProgress
    Status:                      False
    Type:                        Completed
    Last Probe Time:             2021-05-12T09:06:30Z
    Last Transition Time:        2021-05-12T09:06:30Z
    Message:                     The installation has not failed
    Reason:                      InstallationNotFailed
    Status:                      False
    Type:                        Failed
    Last Probe Time:             2021-05-12T09:06:30Z
    Last Transition Time:        2021-05-12T09:06:30Z
    Message:                     The installation is in progress
    Reason:                      InstallationNotStopped
    Status:                      False
    Type:                        Stopped
  Connectivity Majority Groups:  {"192.168.126.0/24":[],"192.168.141.0/24":[]}

```

File: docs/kube-api-conditions.md (hunk 1 above)
Comment (avishayt) [reply docs/kube-api-conditions.md:34]: I considered installing only when doing the actual installation.

Comment (rollandf) [reply docs/kube-api-conditions.md:34]: So we need another reason?
|Stopped|False|InstallationNotStarted|The installation is in progress|If the cluster status is (before installation statuses)|

Comment (avishayt) [reply docs/kube-api-conditions.md:34]: Hmm actually I think we can include it in InstallationNotStopped

Comment (rollandf) [reply docs/kube-api-conditions.md:34]: OK, so what should be changed? The message for InstallationNotStopped ?

Comment (avishayt) [reply docs/kube-api-conditions.md:34]: We can change it to "The installation is waiting to start or in progress"

Comment (rollandf) [reply docs/kube-api-conditions.md]: Done

File: internal/controller/controllers/conditions.go [hunk 2]
```diff
@@ -67,6 +67,22 @@ const (
...
+	ClusterFailedCondition string = hivev1.ClusterInstallFailed
+	ClusterFailedReason    string = "InstallationFailed"
+	ClusterFailedMsg       string = "The installation failed:"
+	ClusterNotFailedReason string = "InstallationNotFailed"
+	ClusterNotFailedMsg    string = "The installation has not failed"
+
+	ClusterStoppedCondition       string = hivev1.ClusterInstallStopped
+	ClusterStoppedFailedReason    string = "InstallationFailed"
+	ClusterStoppedFailedMsg       string = "The installation has stopped due to error"
+	ClusterStoppedCanceledReason  string = "InstallationCancelled"
```
Comment (danielerez) [inline internal/controller/controllers/conditions.go:79]: nit: stick with canceled or cancelled for keeping convention. can be tackled later on.

File: internal/controller/controllers/conditions.go [hunk 3]
```diff
@@ -67,6 +67,22 @@ const (
...
+	ClusterStoppedCanceledMsg     string = "The installation has stopped because it was cancelled"
+	ClusterStoppedCompletedReason string = "InstallationCompleted"
+	ClusterStoppedCompletedMsg    string = "The installation has stopped because it completed successfully"
```
Comment (danielerez) [inline internal/controller/controllers/conditions.go:82]: nit: maybe 'it was completed' sounds better?

**Pull request merged**